### PR TITLE
[Vue] wait for angular to be initialized to post events to avoid loading race condition

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

events can sometimes be posted before angular is ready. Waiting for angular seems to fix the issue.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by sgiehl — APPROVED** [review]

**Comment by sgiehl** [comment]
seems the vue files weren't updated for some reason. Will merge in latest 4.x-dev, maybe that triggers the build.

**Comment by sgiehl** [comment]
@diosmosis I guess I know why the vue files didn't built. The action only ran on `synchronize`, but seems that isn't triggered when the PR is opened. As you didn't push any further commits afterwards it was never triggered. I've added a commit so the action should also run when a PR is opened. Feel free to have a look and merge if you think that's fine.

**Comment by diosmosis** [comment]
@sgiehl I see, thanks for pushing a fix!

File: plugins/CoreHome/vue/src/Matomo/Matomo.ts [hunk 1]
```diff
@@ -65,8 +65,10 @@ piwik.postEvent = function postMatomoEvent(
   piwik.postEventNoEmit(eventName, ...args);
 
   // required until angularjs is removed
-  const $rootScope = piwik.helper.getAngularDependency('$rootScope') as any; // eslint-disable-line
-  return $rootScope.$oldEmit(eventName, ...args);
+  angular.element(() => {
```
Comment (sgiehl) [inline plugins/CoreHome/vue/src/Matomo/Matomo.ts:68]: @diosmosis seems this broke the js tests. See https://app.travis-ci.com/github/matomo-org/matomo/jobs/547700442#L2281-L2291
does there maybe need to be a `return` before the `angular.element`? Or the return type needs to be adjusted

Comment (diosmosis) [reply plugins/CoreHome/vue/src/Matomo/Matomo.ts:68]: No, we can't return anything here unfortunately. It doesn't seem like the result is used in angular.js so it should be fine. I've included it in another PR, but will extract the fix into a new one.

Comment (sgiehl) [reply plugins/CoreHome/vue/src/Matomo/Matomo.ts:68]: Not returning anything is fine as well. Just wasn't sure how this needs to be fixed

**Pull request merged**